controller/volume_controller.go

```diff
@@ -2952,6 +2953,95 @@ func (vc *VolumeController) ResolveRefAndEnqueue(namespace string, ref *metav1.O
 	vc.enqueueVolume(volume)
 }
 
+// createRecurringJobNotExistAndLabel add recurring jobs and group back to a restoring volume and create recurring jobs if not exist.
+func createRecurringJobNotExistAndLabel(log *logrus.Entry, ds *datastore.DataStore, v *longhorn.Volume, jobName string, job *longhorn.RecordVolumeRecurringJob) error {
+	if existJob, err := ds.GetRecurringJob(jobName); err != nil {
+		// create the recurring job if not exist
+		if apierrors.IsNotFound(err) {
+			log.Debugf("Create the recurring job %v when restoring recurring jobs", jobName)
+			if _, err := ds.CreateRecurringJob(&longhorn.RecurringJob{ObjectMeta: metav1.ObjectMeta{Name: jobName}, Spec: job.JobSpec}); err != nil {
+				log.WithError(err).Warnf("Create recurring job %v", jobName)
+				return err
+			}
+		} else {
+			log.WithError(err).Warnf("Get recurring job %v", jobName)
+			return err
+		}
+	} else {
+		// create teh recurring job if job name is used and configuration is different.
+		if !reflect.DeepEqual(existJob.Spec, job.JobSpec) {
+			newJobName := util.RandomID()
+			if _, err := ds.GetRecurringJob(newJobName); err != nil {
+				if apierrors.IsNotFound(err) {
+					newJobSpec := job.JobSpec.DeepCopy()
+					newJobSpec.Name = newJobName
+					log.Debugf("Re-create the recurring job %v instead of a existing job: %v when restoring recurring jobs", newJobName, jobName)
+					if _, err := ds.CreateRecurringJob(&longhorn.RecurringJob{ObjectMeta: metav1.ObjectMeta{Name: newJobName}, Spec: *newJobSpec}); err != nil {
+						log.WithError(err).Warnf("Create recurring job %v", jobName)
+						return err
+					}
+					jobName = newJobName
+				}
+			}
+		}
+	}
+
+	// set Volume recurring jobs/groups
+	if job.FromJob {
+		labelKey := types.GetRecurringJobLabelKey(types.LonghornLabelRecurringJob, jobName)
+		if value, exist := v.Labels[labelKey]; !exist || value == types.LonghornLabelValueEnabled {
+			// disabled recurring jobs/group and it would be enabled after the backup restoration is completed.
```

Comment:
Maybe we can blindly set it then skip the job creation if the volume restoration is in-progress.

Reply:
Modified. Thanks